### PR TITLE
Add ability to set Name of EventBridge Rule

### PR DESCRIPTION
**Pull request by sebastiankasprzak**
*Issue #, if available:*
#2508 

*Description of changes:*
Added `RuleName` Property to `EventBridgeRule` resource which translates to `Name` property in `AWS::Events::Rule`

*Description of how you validated changes:*
Used bin/sam-translate.py to generate a CFN template from SAM template which I then successfully deployed in my AWS account with desired changes taking effect.

*Checklist:*

- [X] Add/update [unit tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions) using:
    - [X] Correct values
    - [ ] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [ ] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)
- [X] `make pr` passes
- [ ] Update documentation
- [X] Verify transformed template deploys and application functions as expected
- [ ] Do these changes include any template validations?
    - [ ] Did the newly validated properties support intrinsics prior to adding the validations? (If unsure, please review [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) before proceeding).
        - [ ] Does the pull request ensure that intrinsics remain functional with the new validations?

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/validator/sam_schema/schema.json [hunk 1]
```diff
@@ -371,6 +371,12 @@
         "Pattern": {
           "type": "object"
         },
+        "RuleName": {
+          "type": "string"
+        },
+        "State": {
+          "type": "string"
+        },
```
Comment (hoffa) [inline samtranslator/validator/sam_schema/schema.json:379]: Is this supposed to be here?

Comment (sebastiankasprzak) [reply samtranslator/validator/sam_schema/schema.json]: leftover from other PR, I removed that commit from this branch/PR

**Review by aahung — COMMENTED** [review]
Thanks for the PR! Same as the previous PR, 
- Can we have some transform tests?
- Can we support the SFN's `Schedule` as well to make them consistent?

**Comment by sebastiankasprzak** [comment]
@aahung transform tests added

**Review by aahung — APPROVED** [review]

**Review by hoffa — APPROVED** [review]

**Comment by aahung** [comment]
Resolved conflicts, will merge once CI checks pass

**Pull request merged**